src/Microsoft.TestPlatform.CrossPlatEngine/Execution/RunTestsWithSources.cs

```diff
@@ -203,5 +207,25 @@ private static string TestCaseFilterToShow(string testCaseFilter)
 
             return testCaseFilterToShow;
         }
+
+        /// <summary>
+        /// Sends Session-End event on in-proc datacollectors
+        /// </summary>
+        protected override void SendSessionEnd()
+        {
+            this.testCaseEventsHandler?.SendSessionEnd();
+        }
+
+        /// <summary>
+        /// Sends Session-Start event on in-proc datacollectors
+        /// </summary>
+        protected override void SendSessionStart()
+        {
+            // Send session start with test sources in property bag for session start event args.
```

Comment:
if (this.testCaseEventsHandler == null) 
{
  return;
}

Remove `?` from `this.testCaseEventsHandler?` after doing so